action.yml

```diff
@@ -63,10 +67,15 @@ runs:
         esac
 
         filename="${filename_prefix}_${filename_suffix}"
-        # cert="${filename}.pem" TODO: verify using cosign
-        # sig="${filename}.sig" TODO: verify using cosign
+        cert="${filename}.pem"
+        sig="${filename}.sig"
 
         curl -sL https://github.com/gittuf/gittuf/releases/download/v${{ inputs.gittuf-version }}/${filename} -o $HOME/.gittuf/$executable_name
+        curl -sL https://github.com/gittuf/gittuf/releases/download/v${{ inputs.gittuf-version }}/${cert} -o $HOME/.gittuf/$cert_name
+        curl -sL https://github.com/gittuf/gittuf/releases/download/v${{ inputs.gittuf-version }}/${sig} -o $HOME/.gittuf/$sig_name
+
+        cosign verify-blob $HOME/.gittuf/$executable_name --certificate $HOME/.gittuf/$cert_name --certificate-identity "https://github.com/gittuf/gittuf/.github/workflows/release.yml@refs/tags/v${{ inputs.gittuf-version }}" --certificate-oidc-issuer "https://token.actions.githubusercontent.com" --signature $HOME/.gittuf/$sig_name
```

Comment:
Should we explicitly check this returns 0?

Reply:
Looks like no! I tried by hardcoding the same sig file for all binaries.

Reply:
Interestingly, I didn't have `set -e` before. I wonder if github enables that by default in this context?